test/unit/org/apache/cassandra/db/SSTableAndMemTableDigestMatchTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db;
+
+import java.nio.ByteBuffer;
+import java.util.NavigableSet;
+import java.util.function.Function;
+
+import com.google.common.collect.Sets;
+import org.junit.Test;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.cql3.ColumnIdentifier;
+import org.apache.cassandra.db.filter.ClusteringIndexNamesFilter;
+import org.apache.cassandra.db.filter.ColumnFilter;
+import org.apache.cassandra.db.marshal.Int32Type;
+import org.apache.cassandra.db.partitions.SingletonUnfilteredPartitionIterator;
+import org.apache.cassandra.db.rows.UnfilteredRowIterator;
+import org.apache.cassandra.net.MessagingService;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+import static org.junit.Assert.assertEquals;
+
+public class SSTableAndMemTableDigestMatchTest extends CQLTester
+{
+    private final static long writeTime = System.currentTimeMillis() * 1000L;
+
+    @Test
+    public void testSelectAllColumns() throws Throwable
+    {
+        testWithFilter(cfs -> ColumnFilter.all(cfs.metadata));
+    }
+
+    @Test
+    public void testSelectNoColumns() throws Throwable
+    {
+        testWithFilter(cfs -> ColumnFilter.selection(cfs.metadata, PartitionColumns.NONE));
+    }
+
+    @Test
+    public void testSelectEmptyColumn() throws Throwable
```

Comment:
Based on the name it seems that the `v2` column should not be populated by the INSERT query.

Reply:
ah, yes, good catch, I'll change that